NETDefault/Gateway.Buckaroo/Mappers/BuckarooRequestMapper.cs

```diff
@@ -0,0 +1,239 @@
+                GroupType = "Address",
+                GroupID = "",
+                Value = request.Debtor.Address.HouseNumberSuffix
+            },
+            new ParameterDto
+            {
+                Name = "ZipCode",
+                GroupType = "Address",
+                GroupID = "",
+                Value = request.Debtor.Address.PostalCode
+            },
+            new ParameterDto
+            {
+                Name = "City",
+                GroupType = "Address",
+                GroupID = "",
+                Value = request.Debtor.Address.City
+            },
+            new ParameterDto
+            {
+                Name = "Country",
+                GroupType = "Address",
+                GroupID = "",
+                Value = "NL"
+            },
+            new ParameterDto
+            {
+                Name = "Email",
+                GroupType = "Email",
+                GroupID = "",
+                Value = request.Debtor.EmailAddress
+            },
+            new ParameterDto
+            {
+                Name = "Mobile",
+                GroupType = "Phone",
+                GroupID = "",
+                Value = request.Debtor.PhoneNumber
+            }
+        };
+
+        var subscriptionRequestDto = new CreateCombinedSubscriptionRequestDto
+        {
+            Description = request.Subscription.Name,
+            Currency = "EUR",
+            StartRecurrent = "true",
+            ContinueOnIncomplete = "1",
+            AmountDebit = 10.00m,
+            AmountCredit = 0,
+            Invoice = request.Subscription.InvoiceDescription ?? "",
+            CustomParameters = customParameters,
+            Services = new ServiceDto
+            {
+                ServiceList = new List<ServiceListDto>
+                    {
+                        new ServiceListDto
+                        {
+                            Name = "Subscriptions",
+                            Action = "CreateCombinedSubscription",
+                            Parameters = new List<ParameterDto>
+                            {
+                                new ParameterDto
+                                {
+                                    Name = "IncludeTransaction",
+                                    GroupType = "",
+                                    GroupID = "",
+                                    Value = "true"
+                                },
+                                new ParameterDto
+                                {
+                                    Name = "ConfigurationCode",
+                                    Value = request.Subscription.ConfigurationCode
+                                }
+                            }
+                        }
+                    }
+            }
+        };
+
+        foreach (var parameter in debtorParameters)
+        {
+            subscriptionRequestDto.Services.ServiceList.FirstOrDefault()!.Parameters!.Add(parameter);

```

Comment:
`FirstOrDefault()` with a `!` is asking for problems. Use a `First()` or `Single()`.